### PR TITLE
ARROW-13959: [R] Update tests for extracting components from date32 objects 

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

File: r/tests/testthat/test-dplyr-lubridate.R [hunk 1]
```diff
@@ -32,97 +32,132 @@ if (tolower(Sys.info()[["sysname"]]) == "windows") {
...
-      select(x)
+test_df <- tibble::tibble(
+  datetime = test_date,
+  date = as.Date("2021-09-09")
+)
+
+test_that("extract year from timestamp", {
+  expect_dplyr_equal(
+    input %>%
+      mutate(x = year(datetime)) %>%
```
Comment (nealrichardson) [inline r/tests/testthat/test-dplyr-lubridate.R:44]: It looks like none of these tests are testing the `date32` path. Shouldn't they all look something like this?

```suggestion
      mutate(
        year(datetime),
        year(date)
      ) %>%
```

File: r/tests/testthat/test-dplyr-lubridate.R [hunk 2]
```diff
@@ -32,97 +32,132 @@ if (tolower(Sys.info()[["sysname"]]) == "windows") {
...
-      mutate(x = year(date)) %>%
-      collect() %>%
-      select(x),
-    test_df %>%
-      mutate(x = year(date)) %>%
-      collect() %>%
-      select(x)
+test_df <- tibble::tibble(
+  datetime = test_date,
+  date = as.Date("2021-09-09")
```
Comment (nealrichardson) [inline r/tests/testthat/test-dplyr-lubridate.R:38]: Wouldn't hurt to confirm the missing data behavior is consistent.

```suggestion
  datetime = c(test_date, NA),
  date = c(as.Date("2021-09-09"), NA)
```

**Review by nealrichardson — CHANGES_REQUESTED** [review]

File: r/tests/testthat/test-dplyr-lubridate.R (hunk 1 above)
Comment (thisisnic) [reply r/tests/testthat/test-dplyr-lubridate.R:44]: So the earlier tests in the file are testing datetimes aka timestamps whereas the later ones are testing dates/date32 objects.  I've added in a couple of comments to make this more clear.

**Review by nealrichardson — APPROVED** [review]
Thanks, I'm ok with this now. One thought: we could consolidate all of these tests to a single test really, just one big `mutate` with a column derived for each of the supported functions. At this point they're just to prevent regressions, so it might be more readable if were more concise.

**Pull request closed without merging**